src/crypto/cbor.cpp

```diff
@@ -0,0 +1,480 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "crypto/cbor.h"
+
+#include <algorithm>
+#include <iomanip>
+#include <sstream>
+
+#define FMT_HEADER_ONLY
+#include <fmt/format.h>
+
+extern "C"
+{
+#include "evercbor/CBORNondet.h"
+}
+
+using namespace ccf::cbor;
+
+namespace
+{
+  Value consume(cbor_nondet_t cbor);
+
+  // Helper to wrap operations with context-aware error messages
+  template <typename F>
+  decltype(auto) with_context(
+    std::string_view context, std::string_view operation, F&& func)
+  {
+    try
+    {
+      return func();
+    }
+    catch (const CBORDecodeError& e)
+    {
+      if (!context.empty())
+      {
+        throw CBORDecodeError(
+          fmt::format("Failed to {} {}: {}", operation, context, e.what()));
+      }
+      throw;
+    }
+  }
+
+  void print_indent(std::ostringstream& os, size_t indent)
+  {
+    for (size_t i = 0; i < indent; ++i)
+    {
+      os << "  ";
+    }
+  }
+
+  Value consume_unsigned(cbor_nondet_t cbor)
+  {
+    Unsigned value{0};
+    if (!cbor_nondet_read_uint64(cbor, &value))
+    {
+      throw CBORDecodeError("Failed to decode unsigned value");
+    }
+    return std::make_unique<WrappedValue>(value);
+  }
+
+  Value consume_signed(cbor_nondet_t cbor)
+  {
+    Signed value{0};
+    if (!cbor_nondet_read_int64(cbor, &value))
+    {
+      throw CBORDecodeError("Failed to decode signed value");
+    }
+    return std::make_unique<WrappedValue>(value);
+  }
+
+  Value consume_byte_string(cbor_nondet_t cbor)
+  {
+    uint8_t* data = nullptr;
+    uint64_t length = 0;
+    if (!cbor_nondet_get_byte_string(cbor, &data, &length))
+    {
+      throw CBORDecodeError("Failed to decode byte string");
+    }
+    Bytes value{data, static_cast<size_t>(length)};
+    return std::make_unique<WrappedValue>(value);
+  }
+
+  Value consume_text_string(cbor_nondet_t cbor)
```

Comment:
It may be tempting to use the CDDL type names (tstr, bstr etc) because they are compact and recognisable if you work with CBOR